src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -1382,31 +1397,47 @@ MavlinkReceiver::handle_message_set_attitude_target(mavlink_message_t *msg)
 					}
 
 					if (!offboard_control_mode.ignore_thrust) { // dont't overwrite thrust if it's invalid
-						// Fill correct field by checking frametype
-						// TODO: add as needed
-						switch (_mavlink->get_system_type()) {
-						case MAV_TYPE_GENERIC:
-							break;
-
-						case MAV_TYPE_FIXED_WING:
-							att_sp.thrust_body[0] = set_attitude_target.thrust;
-							break;
+						if (!_is_vtol) {
+							// Fill correct field by checking frametype
+							// TODO: add as needed
+							switch (_mavlink->get_system_type()) {
+							case MAV_TYPE_GENERIC:
+								break;
+
+							case MAV_TYPE_FIXED_WING:
+								att_sp.thrust_body[0] = set_attitude_target.thrust;
+								break;
+
+							case MAV_TYPE_QUADROTOR:
+							case MAV_TYPE_HEXAROTOR:
+							case MAV_TYPE_OCTOROTOR:
+							case MAV_TYPE_TRICOPTER:
+							case MAV_TYPE_HELICOPTER:
+								att_sp.thrust_body[2] = -set_attitude_target.thrust;
+								break;
+
+							case MAV_TYPE_GROUND_ROVER:
+								att_sp.thrust_body[0] = set_attitude_target.thrust;
+								break;
+							}
+
+							_att_sp_pub.publish(att_sp);
 
-						case MAV_TYPE_QUADROTOR:
-						case MAV_TYPE_HEXAROTOR:
-						case MAV_TYPE_OCTOROTOR:
-						case MAV_TYPE_TRICOPTER:
-						case MAV_TYPE_HELICOPTER:
-							att_sp.thrust_body[2] = -set_attitude_target.thrust;
-							break;
-
-						case MAV_TYPE_GROUND_ROVER:
-							att_sp.thrust_body[0] = set_attitude_target.thrust;
-							break;
+						} else {
```

Comment:
This seems like it is valid for all vtol types. Can't we include it in the switch case with the types for vtols rather than introducing a new boolean variable `_is_vtol`?

Reply:
Done